std/asyncio/system/Process.hx

```diff
@@ -0,0 +1,78 @@
+package asyncio.system;
+
+import haxe.ds.ReadOnlyArray;
+import haxe.io.Bytes;
+import haxe.NoData;
+import haxe.errors.NotImplemented;
+import haxe.Callback;
+
+/**
+	Process execution API
+**/
+class Process {
+	/**
+		Current process handle.
+		Can be used to communicate with the parent process and for self-signalling.
+	**/
+	static public var current(get,never):CurrentProcess;
+	static function get_current():CurrentProcess throw new NotImplemented();
+
+	/**
+		Process id.
+	**/
+	public final pid:Int;
+
+	/**
+		Initial IO streams opened for this process.
+		The first three indices always are:
+		- 0 - stdin
+		- 1 - stdout
+		- 2 - stderr
+		Indices from 3 and higher contain handlers for streams created as configured
+		by the corresponding indices in `options.stdio` field of `options` argument
+		for `asyncio.system.Process.open` call.
+		@see asyncio.system.ProcessOptions.stdio
+	**/
+	public var stdio(get,never):ReadOnlyArray<Stream>;
+	function get_stdio():ReadOnlyArray<Stream> throw new NotImplemented();
+
+	//TODO: this is a dummy constructor to make the compiler shut up about uninitialized finals.
+	function new() {
+		pid = -1;
+	}
+
+	/**
+		Execute `command` with `args` command line arguments, wait for the command
+		to fully finish and invoke `callback` with the exit code and the contents
+		of stdout, and stderr.
+
+		In case the command didn't emit anything to stdout or stderr, the respective
+		field of the result structure will be `null`.
+
+		@see asyncio.system.ProcessOptions for various process configuration options.
+	 */
+	static public function execute(command:String, args:Array<String>, options:ProcessOptions, callback:Callback<Null<{?stdout:Bytes, ?stderr:Bytes, exitCode:Int}>>) {
```

Comment:
A way to specify a `stdin` for the process would be useful. Also I think `args` here and in `open` should be `ProcessOptions`?

Reply:
Moved `args` to `ProcessOptions`.
Stdin is already possible to provie via `ProcessOptions.stdio`.

Reply:
Actually this method is not very well though out. I think it is meant to be a "shortcut" for spawning a process and waiting for its exit code. But you cannot actually interact with its stdin (or any incoming pipes), because in `ProcessOptions.stdio` you only specify that a pipe should be created, not what its contents are. Sending `Bytes` to a process is only possible via an instance of `Process` (or `ChildProcess`), but this method does not expose that.

Reply:
I think that's the cost of shortcut. If one needs to interact with the process, `Process.open` has to be used.

Reply:
@ncannasse I'll move everything to `asys.native` once the API is settled.